### PR TITLE
Fix C FFI extension after moving TH to C++

### PR DESCRIPTION
**Pull request by yf225**
This fixes the issue mentioned in https://github.com/pytorch/pytorch/pull/4041.

### CODE REVIEW

**Review by soumith — APPROVED** [review]

**Comment by yf225** [comment]
gcc 4.8 CI will pass after we merge https://github.com/pytorch/extension-ffi/pull/16.

File: aten/src/TH/THRandom.h [hunk 1]
```diff
@@ -3,33 +3,12 @@
...
-} THGeneratorState;
-
-/* A THGenerator contains all the state required for a single random number stream */
-typedef struct THGenerator {
-  std::mutex mutex; /* mutex for using this generator */
-  THGeneratorState gen_state;
-} THGenerator;
+typedef struct THGenerator THGenerator;
+
+typedef struct THGeneratorState THGeneratorState;
```
Comment (ezyang) [inline aten/src/TH/THRandom.h]: Better comment why struct definition is not here

File: aten/src/TH/THGenerator.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  int normal_is_valid; /* = 0; */
+};
+
+/* A THGenerator contains all the state required for a single random number stream */
+struct THGenerator {
+  std::mutex mutex; /* mutex for using this generator */
+  THGeneratorState gen_state;
+};
+
+#endif
```
Comment (ezyang) [inline aten/src/TH/THGenerator.h]: Add a trailing newline pretty please :)

**Review by ezyang — APPROVED** [review]
Legit.

**Comment by nliz77** [comment]
Thz

**Pull request merged**